packages/in_app_purchase/in_app_purchase_ios/ios/Classes/FIAObjectTranslator.m

```diff
@@ -193,4 +208,24 @@ + (NSDictionary *)getMapFromSKStorefront:(SKStorefront *)storefront
   return map;
 }
 
++ (SKPaymentDiscount *)getSKPaymentDiscountFromMap:(NSDictionary *)map {
+  if (!map) {
+    return nil;
+  }
+
+  NSString *identifier = map[@"identifier"];
```

Comment:
what happens if the `map` is empty, or some of the values are NSNull?

Reply:
I haven't tested on that since on the Dart side all fields are mandatory. Since this is only called from Dart side I skipped doing additional tests.

If you feel it is better to do so I will expand the logic.

Reply:
Thanks! the asserts are for testings, for prod app, should we also prevent nil values?